include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -203,13 +213,23 @@ class BaseCombinable
   T mutable my_data;
 
 public:
-  //! prohibit compiler-generated default ctor
-  BaseCombinable() = delete;
 
   RAJA_HOST_DEVICE
-  constexpr BaseCombinable(T init_val, T identity_ = T())
-      : identity{identity_}, my_data{init_val}
+  BaseCombinable()
   {
+    reset(T(), T());
+  }
+
+  RAJA_HOST_DEVICE
+  BaseCombinable(T init_val, T identity_ = T())
+  {
+    reset(init_val, identity_);
+  }
+  
+  RAJA_HOST_DEVICE
+  void reset(T init_val, T identity_){
```

Comment:
this can be made constexpr by using the comma operator along with additional casts to void. At the cost of readability in this one function, we gain constexpr support. I know there are trade offs, but it might be a worthwhile one.

Reply:
The tradeoff here is that we are assuming T has constexpr constructors -- since this isn't always true (here's looking at you, arrays), maybe it wouldn't matter and you can just ignore my comment here :)